main/src/ca/uwaterloo/flix/language/phase/Inliner.scala

```diff
@@ -501,6 +493,42 @@ object Inliner {
     }
   }
 
+  /**
+   * Performs boolean folding on a given binary expression with the logic:
+   * Only fold if the expression removed is pure, and
+   * For Or-expressions,
+   * fold into true if either the left or right expression is true.
+   * if either the left or right expression is false, fold into the other expression.
+   * For And-expressions,
+   * fold into false, if either the left or right expression is false.
+   * if either the left or right expression is true, fold into the other expression.
+   */
+  private def binaryFold(exp0: LiftedAst.Expression.Binary): LiftedAst.Expression = {
```

Comment:
This function should *not* take a binary expression (as explained above), but rather take these arguments:

sop, op, e1, e2, tpe, purity, loc